### PR TITLE
[INFRA-588] maven.j.o => repo.j.o

### PR DESCRIPTION
**Pull request by batmat**
URL have been changed quite a long time ago, and started to fail badly recently. Time to fix those.

Ping @rtyler

### CODE REVIEW

**Comment by batmat** [comment]
When agreed & merged, I'll file the same PR on https://github.com/jenkinsci/jenkins/blob/master/pom.xml#L64L73 and https://github.com/jenkinsci/plugin-pom/blob/master/pom.xml#L734L744

Or possibly better, just release the org pom, and then use it in those projects + remove the override, BTW.

Also : 
ping @omehagan and @dalvizu as commenters of [INFRA-588](https://issues.jenkins-ci.org/browse/INFRA-588)

**Comment by rtyler** [comment]
Pinging @omehegan for this :)

File: pom.xml [hunk 1]
```diff
@@ -103,11 +103,11 @@
     <repository>
       <uniqueVersion>false</uniqueVersion>
       <id>maven.jenkins-ci.org</id>
-      <url>http://maven.jenkins-ci.org:8081/content/repositories/releases</url>
+      <url>http://repo.jenkins-ci.org/releases/</url>
```
Comment (rtyler) [inline pom.xml]: Is there a compelling reason not to use HTTPs by default here? https://repo.jenkins-ci.org has a proper certificate and everything

**Comment by omehegan** [comment]
It worked for me to do a release yesterday, so I say +1.

**Comment by batmat** [comment]
Ping @jenkinsci/code-reviewers for more eyes. Very few lines, but quite important.

**Comment by daniel-beck** [comment]
Also wondering about HTTPS.

**Comment by batmat** [comment]
Right. I'll try releasing a beta version of one of my plugins tonight, to
check it works at least for me, and if so will update the PR with HTTPS
instead.

**Comment by dalvizu** [comment]
:+1: 
+1 to https - I used that in my work-around just fine

**Comment by batmat** [comment]
@daniel-beck @rtyler HTTP->HTTPS done & tested on https://github.com/jenkinsci/radiatorview-plugin/tree/radiatorviewplugin-1.27-alpha-2-dontuse => http://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/radiatorviewplugin/1.27-alpha-2-dontuse/

For future potential reference, my release worked fine with:
```
$ mvn -V clean release:prepare release:perform
Apache Maven 3.3.1 (cab6659f9874fa96462afef40fcf6bc033d58c1c; 2015-03-13T21:10:27+01:00)
Maven home: /home/tiste/.tools/Mavens/apache-maven-3.3.1
Java version: 1.7.0_79, vendor: Oracle Corporation
Java home: /home/tiste/.tools/JDKs/jdk1.7.0_79/jre
Default locale: fr_FR, platform encoding: UTF-8
OS name: "linux", version: "4.4.4-301.fc23.x86_64", arch: "amd64", family: "unix"
[INFO] Scanning for projects...
```

**Comment by aheritier** [comment]
+1

**Comment by Vlatombe** [comment]
:+1:

**Comment by abayer** [comment]
Merging and releasing

**Pull request merged**